sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -4348,12 +4348,26 @@ object GpuOverrideUtil extends Logging {
   }
 }
 
+object GpuOverridesContext {
+  private val local = ThreadLocal.withInitial[String](() => null);
+  def set(s: String): Unit = local.set(s)
+
+  def get(): Option[String] = Option(local.get())
+
+  def clear() = local.set(null)
+}
+
 /** Tag the initial plan when AQE is enabled */
 case class GpuQueryStagePrepOverrides() extends Rule[SparkPlan] with Logging {
   override def apply(sparkPlan: SparkPlan): SparkPlan = GpuOverrideUtil.tryOverride { plan =>
     // Note that we disregard the GPU plan returned here and instead rely on side effects of
     // tagging the underlying SparkPlan.
-    GpuOverrides().apply(plan)
+    try {
+      GpuOverridesContext.set("AQE Query Stage Prep")
+      GpuOverrides().apply(plan)
```

Comment:
It seems odd that we're using a thread local to work around the lack of a parameter to pass when we're the ones controlling the code doing the calling and the code being called.  Why not add a form of `GpuOverrides.apply` that takes an extra context string argument rather than poking a thread local and looking it up later?